primitives/sidechain/src/traits/mod.rs

```diff
@@ -138,28 +137,37 @@ pub trait SignedBlock: Encode + Decode + Send + Sync + Debug + Clone {
 	fn verify_signature(&self) -> bool;
 }
 
-/// Provide signing logic blanket implementations for all block types satisfying the trait bounds.
-pub trait SignBlock<
-	SidechainBlock: Block,
-	SignedSidechainBlock: SignedBlock<Block = SidechainBlock>,
->
-{
-	fn sign_block<P: Pair>(self, signer: &P) -> SignedSidechainBlock
-	where
-		<SignedSidechainBlock as SignedBlock>::Signature: From<<P as sp_core::Pair>::Signature>;
-}
+#[cfg(feature = "full_crypto")]
+pub use crypto::*;
+
+#[cfg(feature = "full_crypto")]
+mod crypto {
```

Comment:
Signing logic is only compiled with `full-crypto`. We do not need it in the runtime, and we don't want all of these dependencies there.